### PR TITLE
GitHub Issue NOAA-EMC/GSI#265. MinMon maintenance update.

### PR DESCRIPTION
**Pull request by EdwardSafford-NOAA**
General MinMon maintenance update including:

- Remove numbered versions, condense to gdas, gfs, and shared directories.
- Add support for additional arguments to all dev (non-operational) scripts, eliminating the use of "driver" scripts.
- Update test driver scripts.

### CODE REVIEW

**Comment by EdwardSafford-NOAA** [comment]
@MichaelLueken-NOAA if @DavidHuber-NOAA has time I'd welcome his review.

File: util/Minimization_Monitor/data_xtrct/ush/find_cycle.pl [hunk 1]
```diff
@@ -148,7 +148,6 @@
          $hr_ctr = $hr_ctr - 1;
          
          $newdir = "${dirpath}/${sortmm[$ctr]}/${hrs[$hr_ctr]}/${lcm}";
```
Comment (DavidHuber-NOAA) [inline util/Minimization_Monitor/data_xtrct/ush/find_cycle.pl:150]: I must have failed to commit this change when I ported this to S4 and Jet.  To enable plotting data from the global workflow, there should be a check for `$newdir`.  If it doesn't exist, then set `$newdir` to `"${dirpath}/${sortmm[$ctr]}"`.

File: util/Minimization_Monitor/image_gen/ush/MinMon_Plt.sh [hunk 2]
```diff
@@ -45,38 +45,27 @@ do
...
-   . ${minmon_version_file}
-   echo "able to source ${minmon_version_file}"
-else
-   echo "Unable to source ${minmon_version_file} file"
-   exit 2
-fi
-
 minmon_config=${minmon_config:-${top_parm}/MinMon_config}
 if [[ -s ${minmon_config} ]]; then
    . ${minmon_config}
```
Comment (DavidHuber-NOAA) [inline util/Minimization_Monitor/image_gen/ush/MinMon_Plt.sh:71]: Adding a check that the `$minmon_config` and `$minmon_user_setting` were sourced correctly (i.e. `$? = 0`) would make debugging easier.

**Review by DavidHuber-NOAA — CHANGES_REQUESTED** [review]
I gave the plotting routines a test run on S4 and everything ran OK with the changes mentioned for `gdas`.  However, when running with `gfs`, the reduction plot is empty.  Note that this was for a test case where the GFS ran on only the 00Z cycle.
192_80_2_gfs 2020080900 reduction

**Comment by DavidHuber-NOAA** [comment]
Below is the output from grads when running the reduction code:

```
+ /opt/grads/2.2.1/bin/grads -blc 'run 2020080900_plot_reduction.gs'

Grid Analysis and Display System (GrADS) Version 2.2.1
Copyright (C) 1988-2018 by George Mason University
GrADS comes with ABSOLUTELY NO WARRANTY
See file COPYRIGHT for more information

Config: v2.2.1 little-endian readline grib2 netcdf hdf4-sds hdf5 opendap-grids geotiff shapefile
Issue 'q config' and 'q gxconfig' commands for more detailed configuration information
GX Package Initialization: Size = 11 8.5
Running in Batch mode
process source, date 192_80_2_gfs , date
nchan,tlast = 202 29
Low Level I/O Error:  Read error on data file
  Data file name = %y4%m2%d2%h2.reduction.ieee_d
  Error reading 808 bytes at location 0
Data Request Error:  Error for variable 'reduct'
Operation Error:  Error from ave function
  Error ocurred at column 1
DEFINE error:  Invalid expression.
SET Error: parea values beyond page limits
Low Level I/O Error:  Read error on data file
  Data file name = %y4%m2%d2%h2.reduction.ieee_d
  Error reading 808 bytes at location 0
Data Request Error:  Error for variable 'reduct'
  Error ocurred at column 1
DISPLAY error:  Invalid expression
  Expression = reduct
Syntax Error:  Invalid Operand
  'ave1' not a variable or function name
  Error ocurred at column 1
DISPLAY error:  Invalid expression
  Expression = ave1
GX Package Terminated
```

File: util/Minimization_Monitor/data_xtrct/ush/find_cycle.pl (hunk 1 above)
Comment (EdwardSafford-NOAA) [reply util/Minimization_Monitor/data_xtrct/ush/find_cycle.pl:150]: OK, I've added a check for $newdir and set it to "${dirpath}/${sortmm[$ctr]}" if the default location doesn't exist.

**Comment by EdwardSafford-NOAA** [comment]
@DavidHuber-NOAA I've been able to reproduce that on wcoss_d.  Somehow the reduction files are ending up too small, so it's actually an extraction issue.  I'm exploring that now.

**Comment by EdwardSafford-NOAA** [comment]
I figured out what's going on with the reduction plots.  The extracted data files are fine.  The issue is that the expected number of iterations was wrong for gfs data.  Data for the gdas run has 202 iterations while gfs has 152.  Somewhere along the line that logic was lost, probably my error.  Correcting the xdef line in the GrADS reduction.ctl file fixes the issue.  While chasing that down I also found a gfs specific bug in the call to find_cycle.pl and fixed that.

File: util/Minimization_Monitor/image_gen/ush/MinMon_Plt.sh (hunk 2 above)
Comment (EdwardSafford-NOAA) [reply util/Minimization_Monitor/image_gen/ush/MinMon_Plt.sh:71]: Good suggestion.  I've made that change.

**Comment by EdwardSafford-NOAA** [comment]
I've completed all changes/corrections and retested.  @DavidHuber-NOAA please take another look when you have a moment.

**Review by DavidHuber-NOAA — APPROVED** [review]
@EdwardSafford-NOAA Everything looks good to me now and all plots generated as expected.  Thanks!

**Comment by MichaelLueken-NOAA** [comment]
Since @DavidHuber-NOAA has approved this update and there are no changes to the source code.  I will now give final approval to these changes and merge them to the authoritative repository.

**Pull request merged**